### PR TITLE
BUG: SeriesGroupby.nunique raises an IndexError on empty Series #12553

### LINKED ISSUE #12553: SeriesGroupby.nunique raises an IndexError on empty Series
#### Code Sample, a copy-pastable example if possible

``` python
In [18]: b = pandas.Series()

In [19]: g = b.groupby(level = 0)

In [20]: g.nunique()
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-20-fbbfc3108eac> in <module>()
----> 1 g.nunique()

/usr/local/lib/python2.7/dist-packages/pandas/core/groupby.pyc in nunique(self, dropna)
   2693 
   2694         out = np.add.reduceat(inc, idx).astype('int64', copy=False)
-> 2695         return Series(out if ids[0] != -1 else out[1:],
   2696                       index=self.grouper.result_index,
   2697                       name=self.name)

IndexError: index 0 is out of bounds for axis 0 with size 0
```
#### Expected Output

``` python
0
```
#### output of `pd.show_versions()`

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-79-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.17.1
nose: 1.3.1
pip: 8.0.2
setuptools: 20.1.1
Cython: None
numpy: 1.10.4
scipy: 0.13.3
statsmodels: 0.5.0
IPython: 4.1.1
sphinx: None
patsy: 0.2.1
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: None
tables: 3.1.1
numexpr: 2.2.2
matplotlib: 1.3.1
openpyxl: 1.7.0
xlrd: 0.9.2
xlwt: 0.7.5
xlsxwriter: None
lxml: None
bs4: 4.2.1
html5lib: 0.999
httplib2: 0.8
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
Jinja2: None
```

### PR DESCRIPTION
**Pull request by thanasis2028**
Closes #12553
xref #13239 
Made a new pull request to clean up from #12557

### CODE REVIEW

**Comment by jreback** [comment]
needs tests

**Comment by thanasis2028** [comment]
I'm sry what am I supposed to do? Do you want me to run `nosetests pandas` or make new tests?

**Comment by jreback** [comment]
new tests, see contributing docs [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html)

you have a failing case, so create a test which fails w/o your fix (put the tests in test_groupby.py)

then your fix should make it pass

**Comment by jreback** [comment]
can you rebase / update & add whatsnew for 0.18.2

**Comment by jreback** [comment]
can you rebase/update

**Comment by thanasis2028** [comment]
Done, sry for 2 commits but when I try to rebase everything gets messed up...

File: pandas/core/groupby.py [hunk 1]
```diff
@@ -2833,6 +2833,10 @@ def true_and_notnull(x, *args, **kwargs):
     def nunique(self, dropna=True):
         """ Returns number of unique elements in the group """
         ids, _, _ = self.grouper.group_info
+
+        if len(ids) == 0:  # bugfix for 12553
+            return Series([])
```
Comment (jreback) [inline pandas/core/groupby.py]: need ``name=self.name``

File: pandas/tests/test_groupby.py [hunk 2]
```diff
@@ -6391,6 +6391,14 @@ def testit(label_list, shape):
     testit(label_list, shape)
 
 
+def test_nunique_empty_Series():
+    series = Series()
```
Comment (jreback) [inline pandas/tests/test_groupby.py]: try with ``s = Series(name='foo')``

use ``tm.assert_series_equal`` for the comparison

**Comment by jreback** [comment]
can you rebase / update

File: pandas/tests/test_groupby.py [hunk 3]
```diff
@@ -6465,6 +6465,14 @@ def testit(label_list, shape):
     testit(label_list, shape)
 
 
+def test_nunique_empty_Series():
+    series = Series(name='foo')
+    group = series.groupby(level=0)
+    result = group.nunique()
+    expected = Series(name='foo')
+    assert_series_equal(result, expected)
```
Comment (jorisvandenbossche) [inline pandas/tests/test_groupby.py]: Can you put this test inside the TestGroupBy class? Eg after `test_nunique_with_object` at Line 6404 (https://github.com/thanasis2028/pandas/blob/744e537c3c73ecb30a62a7707db138b7f4dafcd8/pandas/tests/test_groupby.py#L6400)

**Comment by jorisvandenbossche** [comment]
@thanasis2028 Your change is causing an error in `test_resample_empty_series` (probably resampling an empty series and calling `unique` on that) 
Can you have a look at that?

**Comment by jorisvandenbossche** [comment]
@thanasis2028 Do you have time to update this PR?

**Comment by jreback** [comment]
can you rebase / update this?

**Comment by jreback** [comment]
this was closed by this PR [here](https://github.com/pandas-dev/pandas/commit/c0e13d1bccd4a783486eba8cc769db48a7875de8)

**Pull request closed without merging**